test/src/test/java/org/corfudb/integration/MergeTestListener.java

```diff
@@ -0,0 +1,204 @@
+
+    private CorfuStore corfuStoreSink;
+
+    private final String namespace = "LR-Test";
+
+    // Updates received through streaming
+    @Getter
+    public static LinkedList<CorfuStreamEntry> updates = new LinkedList<>();
+
+    public Set<Sample.StringKey> deletedKeys = new HashSet<>();
+
+    public Map<Sample.StringKey, CorfuStoreEntry> addedKeys = new HashMap<>();
+
+    public Set<Sample.StringKey> existingKeysInMergedTable = new HashSet<>();
+
+
+    public static List<CorfuStoreEntry<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>> entries
+            = new ArrayList<>();
+
+    private final int numTableForGroupA;
+
+    private final int numTableForLocalTables;
+
+    MergeTestListener(CorfuStore corfuStoreSink, String namespace, int numTableForGroupA, int numTableForLocalTables) {
+        super(corfuStoreSink, namespace);
+        this.corfuStoreSink = corfuStoreSink;
+        this.numTableForGroupA = numTableForGroupA;
+        this.numTableForLocalTables = numTableForLocalTables;
+    }
+
+    @Override
+    public void onError(Throwable throwable) {
+        log.error("Error on listener", throwable);
+        fail("onError for LRTestListener: " + throwable.toString());
+    }
+
+    /**
+     * Invoke when snapshot sync starts
+     */
+    @Override
+    protected void onSnapshotSyncStart() {
+        log.info("Snapshot sync started");
+    }
+
+    /**
+     * Apply addedKeys and deletedKeys to merged table, add keys to
+     * existingKeysInMergedTable from addedKeys and deletedKeys.
+     */
+    @Override
+    protected void onSnapshotSyncComplete() {
+        try (TxnContext tx = corfuStoreSink.txn(NAMESPACE)){
+            existingKeysInMergedTable.addAll(addedKeys.keySet());
+            addedKeys.forEach((addedKey, e) -> {
+                tx.putRecord(tx.getTable(MERGED_TABLE_NAME), addedKey,
+                        SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString().replaceAll("[^0-9]", "")).build(), e.getMetadata());
+            });
+
+            existingKeysInMergedTable.removeAll(deletedKeys);
+            deletedKeys.forEach(deletedKey -> {
+                tx.delete(tx.getTable(MERGED_TABLE_NAME), deletedKey);
+            });
+
+            tx.commit();
+        }
+
+        log.info("Snapshot sync complete");
+    }
+
+    /**
+     * Write local tables to merged table, process updates and add the keys to addedKeys
+     * and deletedKeys for replicated table.
+     *
+     * @param results Entries received in a single transaction as part of a snapshot sync
+     */
+    @Override
+    protected void processUpdatesInSnapshotSync(CorfuStreamEntries results) {
+        log.info("Processing updates in snapshot sync");
+        results.getEntries().forEach((schema, entries) -> {
+            entries.forEach(e -> {
+                Sample.StringKey key = (Sample.StringKey) e.getKey();
+                if (schema.getTableName().contains("Local_Table00")) {
```

Comment:
Please declare the prefix in a String field

Reply:
also, why initialize `key` variable out of the if clause? seems confusing for me

Reply:
For the first comment, just imported a local table prefix and used that in line 115!

For the second comment, I think the key of entry is used frequently, so I initialize `key` here, then I do not need to invoke `getKey` every time. I am okay to remove this initialization if you find this is confusing to read.